share/completions/gradle.fish

```diff
@@ -37,3 +37,25 @@ complete -c gradle -l continuous -s t -d 'Enable continuous build'
 complete -c gradle -l no-search-upward -s u -d 'Don\'t search in parent folders for settings file'
 complete -c gradle -l version -s v -d 'Print version'
 complete -c gradle -l exclude-task -s x -x -d 'Specify task to be excluded from execution'
+
+# https://github.com/hanny24/gradle-fish/blob/master/gradle.load
+function __cache_or_get_gradle_completion
+  # Set up cache directory
+  if test -z $XDG_CACHE_HOME
+    set XDG_CACHE_HOME $HOME/.cache/
+  end
+  mkdir -m 700 -p $XDG_CACHE_HOME/gradle-completions
+
+  set -l hashed_pwd (fish_md5 $PWD)
+  set -l gradle_cache_file $XDG_CACHE_HOME/gradle-completions/$hashed_pwd
+  if not command test -f $gradle_cache_file; or command test build.gradle -nt $gradle_cache_file
```

Comment:
The `test -f` can use the builtin - the second one only needs it because our builtin doesn't support "-nt".